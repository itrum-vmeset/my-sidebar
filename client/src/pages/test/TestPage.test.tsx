import { fireEvent, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import TestPage from "./TestPage";

describe("TestPage", () => {
  test("test render", () => {
    render(<TestPage />);
    const testText = screen.getByText(/testpage/i);
    expect(testText).toBeInTheDocument();
    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();
    const input = screen.getByPlaceholderText(/input text/i);
    expect(input).toMatchSnapshot();
    expect(input).toBeInTheDocument();
  });

  test("test toggle block visible", () => {
    render(<TestPage />);
    const toggleBtn = screen.getByTestId("toggleBtn");
    expect(screen.queryByTestId("toggleBlock")).toBeNull();
    fireEvent.click(toggleBtn);
    expect(screen.queryByTestId("toggleBlock")).toBeInTheDocument();
    fireEvent.click(toggleBtn);
    expect(screen.queryByTestId("toggleBlock")).toBeNull();
  });

  test("textarea input", () => {
    render(<TestPage />);
    const textarea = screen.getByTestId("textarea");
    expect(textarea).toContainHTML("");
    fireEvent.input(textarea, {
      target: { value: "123" },
    });
    expect(textarea).toContainHTML("123");
  });
});
