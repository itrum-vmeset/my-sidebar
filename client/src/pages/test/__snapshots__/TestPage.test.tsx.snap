// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestPage test render 1`] = `
<textarea
  data-testid="textarea"
  placeholder="input text..."
/>
`;
